hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -257,10 +296,15 @@ int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
 
                 // GSM/UMTS Specific -------------------------------------------
                 // +UUPSDD: <profile_id>
-                if (sscanf(cmd, "UUPSDD: %d",&a) == 1) {
-                    if (*PROFILE == a) {
+                if (sscanf(cmd, "UUPSDD: %c", s) == 1) {
+                    DEBUG_D("UUPSDD: %s matched\r\n", PROFILE);
```

Comment:
I started with %c but then switched to %s because I already had a buffer defined.  Need to change %c to %s